app/src/main/java/com/spoony/spoony/presentation/register/componet/TopLinearProgressbar.kt

```diff
@@ -0,0 +1,90 @@
+package com.spoony.spoony.presentation.register.componet
+
+import androidx.compose.animation.core.Animatable
+import androidx.compose.animation.core.FastOutSlowInEasing
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.LinearProgressIndicator
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.StrokeCap
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.spoony.spoony.core.designsystem.theme.SpoonyAndroidTheme
+
+@Composable
+fun TopLinearProgressBar(
+    currentStep: Int
+) {
+    val progress = remember { Animatable(0f) }
+    val targetProgress = currentStep.toFloat() / 3
+
+    LaunchedEffect(currentStep) {
+        progress.animateTo(
+            targetValue = targetProgress,
+            animationSpec = tween(
+                durationMillis = 300,
+                easing = FastOutSlowInEasing
+            )
+        )
+    }
+
+    LinearProgressIndicator(
+        progress = { progress.value },
+        modifier = Modifier
+            .clip(RoundedCornerShape(50))
```

Comment:
P3: CircleShape 로 달면 어떨까요!
```suggestion
            .clip(CircleShape)
```